### PR TITLE
Add customer model

### PR DESCRIPTION
**Pull request by Gataniel**

### CODE REVIEW

File: db/migrate/20150326084125_create_customers.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+class CreateCustomers < ActiveRecord::Migration
+  def change
+    create_table :customers do |t|
+      t.string :name
+      t.references :organization, index: true
+
+      t.timestamps
+    end
+    add_index :customers, :name, unique: true
```
Comment (AlexanderUlitin) [inline db/migrate/20150326084125_create_customers.rb]: @Gataniel I think it should be uniq for one organization only, so index should be for `[:name, :organization_id]`.

File: app/models/customer.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+#
+#  id              :integer          not null, primary key
+#  name            :string(255)
+#  organization_id :integer
+#  created_at      :datetime
+#  updated_at      :datetime
+#
+
+class Customer < ActiveRecord::Base
+  belongs_to :organization
```
Comment (AlexanderUlitin) [inline app/models/customer.rb]: @Gataniel let's add association for transactions too?

File: app/models/transaction.rb [hunk 3]
```diff
@@ -29,6 +31,7 @@ def total
   belongs_to :category, inverse_of: :transactions
   belongs_to :bank_account, inverse_of: :transactions
   has_one :organization, through: :bank_account, inverse_of: :transactions
+  has_one :customer, foreign_key: :id, primary_key: :customer_id
```
Comment (AlexanderUlitin) [inline app/models/transaction.rb]: @Gataniel why we don't use `belongs_to` here?

Comment (Gataniel) [inline app/models/transaction.rb]: @AlexanderUlitin  shall we change `has_one :organization, through: :bank_account` to `belongs_to` as well?

Comment (AlexanderUlitin) [inline app/models/transaction.rb]: @Gataniel let's keep organization as is, but I think we should change it for customer.

File: app/models/customer.rb [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+#  id              :integer          not null, primary key
+#  name            :string(255)      not null
+#  organization_id :integer
+#  created_at      :datetime
+#  updated_at      :datetime
+#
+
+class Customer < ActiveRecord::Base
+  belongs_to :organization, inverse_of: :customers
+  has_many :transactions, inverse_of: :customer
```
Comment (AlexanderUlitin) [inline app/models/customer.rb]: @Gataniel please add validation for existing organization.

Comment (AlexanderUlitin) [inline app/models/customer.rb]: @Gataniel and also validation for uniq name in organization.

Comment (shir) [inline app/models/customer.rb]: @Gataniel and add validation for name field

Comment (AlexanderUlitin) [inline app/models/customer.rb]: @Gataniel also validation for name presence.

File: spec/models/customer_spec.rb [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+#  updated_at      :datetime
+#
+
+require 'spec_helper'
+
+describe Customer do
+  context "association" do
+    it { should belong_to(:organization) }
+    it { should have_many(:transactions) }
+  end
```
Comment (AlexanderUlitin) [inline spec/models/customer_spec.rb]: @Gataniel add specs for validations.

File: db/migrate/20150326084125_create_customers.rb [hunk 6]
```diff
@@ -0,0 +1,11 @@
+class CreateCustomers < ActiveRecord::Migration
+  def change
+    create_table :customers do |t|
+      t.string :name, null: false
+      t.references :organization, index: true, null: false
+
+      t.timestamps
+    end
+    add_index :customers, [:name, :organization_id], unique: true
```
Comment (shir) [inline db/migrate/20150326084125_create_customers.rb]: @Gataniel if you have this unique index then add appropriate validation

File: spec/models/customer_spec.rb [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+#  organization_id :integer
+#  created_at      :datetime
+#  updated_at      :datetime
+#
+
+require 'spec_helper'
+
+describe Customer do
+  context "association" do
+    it { should belong_to(:organization) }
```
Comment (shir) [inline spec/models/customer_spec.rb]: @Gataniel use `is_expected.to` instead `should`

File: app/models/customer.rb (hunk 4 above)
Comment (shir) [inline app/models/customer.rb]: @Gataniel use `dependent:` option

Comment (AlexanderUlitin) [inline app/models/customer.rb]: @shir hay, I've just noticed that you also added this comments)))

File: app/models/customer.rb [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+#  updated_at      :datetime
+#
+
+class Customer < ActiveRecord::Base
+  belongs_to :organization, inverse_of: :customers
+  has_many :transactions, dependent: :destroy, inverse_of: :customer
+
+  validates :name, presence: true
+  validates :organization, presence: true
+  validates :organization_id, uniqueness: { scope: :name }
```
Comment (AlexanderUlitin) [inline app/models/customer.rb]: @Gataniel  @shir I think it should be `validates :name, uniqueness: { scope: :organization_id }`, am I right?

Comment (shir) [inline app/models/customer.rb]: @Gataniel  @AlexanderUlitin sure. It's name uniqueness within organisation.

Comment (Gataniel) [inline app/models/customer.rb]: @AlexanderUlitin @shir What's about
```
class Member < ActiveRecord::Base
  validates :organization_id, uniqueness: { scope: :user_id }
```
is it okey?

Comment (AlexanderUlitin) [inline app/models/customer.rb]: @Gataniel I think no, should be changed too.

Comment (AlexanderUlitin) [inline app/models/customer.rb]: @Gataniel I think no, should be changed too.

**Pull request merged**